src/controllers/objecting.entity.name.controller.ts

```diff
@@ -77,7 +80,20 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
     } else {
       return showPageWithSessionDataIfPresent(session, res);
     }
-    return next(new Error("No Session present"));
+  }
+  return next(new Error("No Session present"));
```

Comment:
we need a test to check the error screen is shown for this line, if we had one it would have failed and highlighted the bug. We need to add a test to prevent the bug from being re-introduced.

Reply:
There is a test for this but as the error hadling is a catch all at the end of the chain it is difficult discern whether it reroutes to the error screen and returns 500 because of any one particular error. If you take out the next line with the "No session present" message then the result is the same.

Reply:
my 'npm run coverage' command is showing line 84 as being un-tested. We just want to ensure that the error page is displayed if we call 'get' with an undefined or null session. If we didn't have the return next(new err..), the browser request would time out as there is no session so nothing would be returned. So, essentially:
**When** the 'get' function is called
**And** the session is missing
**Then** the error screen should be returned

we should be able to look at the text returned in the response to see if it is the error screen.

Reply:
Yes yes yes yes I understand that that's not the problem here.

Reply:
ok, i'll have to look into it a bit later, see if i can sort it

Reply:
A test was written to cover this essential, crucial and very important fact just stated, yet in spite of it all that line is not covered regardless of whether I set the mock not to initailise the session or explicitly set it to undefined.

Reply:
Children, animal and node unit tests hey.

Reply:
ah ok, just thought it hadn't been covered. The tests aren't the easiest, but practice makes perfect !